django/contrib/admin/filters.py

```diff
@@ -236,18 +236,26 @@ def __init__(self, field, request, params, model, model_admin, field_path):
         self.lookup_val = params.get(self.lookup_kwarg)
         self.lookup_val2 = params.get(self.lookup_kwarg2)
         super().__init__(field, request, params, model, model_admin, field_path)
-        if (self.used_parameters and self.lookup_kwarg in self.used_parameters and
+        if (self.used_parameters and
+                self.lookup_kwarg in self.used_parameters and
                 self.used_parameters[self.lookup_kwarg] in ('1', '0')):
             self.used_parameters[self.lookup_kwarg] = bool(int(self.used_parameters[self.lookup_kwarg]))
 
     def expected_parameters(self):
         return [self.lookup_kwarg, self.lookup_kwarg2]
 
     def choices(self, changelist):
-        for lookup, title in (
-                (None, _('All')),
+        if self.field.flatchoices:
+            list_of_choices = [(str(int(x[0])), x[1]) for x in self.field.flatchoices if x[0] is not None]
+        else:
+            list_of_choices = [
                 ('1', _('Yes')),
-                ('0', _('No'))):
+                ('0', _('No')),
+            ]
```

Comment:
IMO we could simplify this with using `dict`, e.g.
```python
def choices(self, changelist):
    choices = {
        None: _('All'),
        True: _('Yes'),
        False: _('No'),
        **{lookup: title for lookup, title in self.field.flatchoices},
    }
    for lookup, title in choices.items():
        lookup_val = str(int(lookup)) if lookup is not None else lookup
        yield {
            'selected': self.lookup_val == lookup_val and not self.lookup_val2,
            'query_string': changelist.get_query_string({self.lookup_kwarg: lookup_val}, [self.lookup_kwarg2]),
            'display': title,
        }
    ...
```

Reply:
Simplified this using `dict`